shortfin/python/shortfin_apps/llm/components/batcher.py

```diff
@@ -69,6 +72,8 @@ def __init__(
 
         self.program_isolation = program_isolation
 
+        self.active_requests: Dict[str, List[LlmInferenceExecRequest]] = {}
```

Comment:
These kinds of stateful components are pretty big red flags to me. We should avoid having under cover mutation where possible. I am pretty sure that by tracking the run command we can handle fine with depending on global mutation.